### DISCUSSION TITLE
Can we use renovate without the renovate.json file

### DISCUSSION THREAD

**Discussion by B3ns44d** [renovatebot/renovate#19947] [Request Help] [answered]
We are trying to configure renovate and i wanted to know if there is a way to configure it without adding the `renovate.json` file to all the repositories?

thanks in advance.

**Comment by viceice** [comment, marked as answer]
👉  https://docs.renovatebot.com/self-hosted-configuration/#requireconfig

**Reply by B3ns44d** [reply]
@viceice thanks, it worked, but I still need to add the configuration(renovate.json file) somewhere so that the runner would take it into account when creating MRs.

**Reply by B3ns44d** [reply]
i don't want it to create the "Configure Renovate" MR, so i've did `RENOVATE_REQUIRE_CONFIG=ignored` it worked but it
didn't take the config(the one in renovate-config) in consideration, normally i would do something like this to extend
the config

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "local>renovate/renovate-config"
  ]
}
```

**Reply by viceice** [reply]
you need to create a `config.js` or `config.json` to the bot config to apply config globally

**Reply by B3ns44d** [reply]
yes indeed i have the `config.js` file that looks something like this:

```js
module.exports = {
    endpoint: 'https://gitlab.com/api/v4/',
    platform: 'gitlab',
    autodiscover: false,
    repositories: [
        "org/repo1",
    ],
};
```

i'm not completely sure about how to add the `renovate.json config` to this config,

**Reply by viceice** [reply]
just add the config options to the exports object. you can use all self-hosted and repo renovate config options there

**Reply by B3ns44d** [reply]
it worked thank you

**Reply by B3ns44d** [reply]
hello, @viceice after some testing, I've discovered that it didn't actually work; all I need to find out what javascript `attribute` to add to the `module.exports = {}` to extend my renovate-config, which is located in `renovate/renovate-config`

this is what i've did in the `config.js`

```javascript
module.exports = {
.... 
config: {
        extends: ["local>renovate/renovate-config"],
    },
....
}
```

and I didn't really take that into consideration, any idea about this please, thanks

**Reply by viceice** [reply]
remove the `config` object and include all settings directly to the parent config

**Reply by B3ns44d** [reply]
well, by using `extends` directly, I was able to fix it, i wanted to include all of the settings in a different repository rather than in the `config.js` file itself,
